### PR TITLE
[nzherald] Add NZHeraldIE

### PR DESCRIPTION
**Pull request by coletdjnz**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [X] At least skimmed through [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) sections
- [X] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [X] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [X] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [ ] Improvement
- [X] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information
This adds supports for videos on nzherald.co.nz (NZ news site)
Upstream request: resolves https://github.com/ytdl-org/youtube-dl/issues/28267

They use Brightcove so this extractor redirects to that once we extract the relevant player url.

From my testing there are at least two page formats for videos:
- video embedded in article
- dedicated video page

~~I notice a few other similar extractors hard code the Brightcove player and account ids. Might be a bit overkill but I've added them as a fallback here and opted to try extract them first~~

Known issues:
- raises unexpected extractor error for articles without a video

### CODE REVIEW

**Comment by pukkandan** [comment]
> I notice a few other similar extractors hard code the Brightcove player and account ids. Might be a bit overkill but I've added them as a fallback here and opted to try extract them first.

I havent looked at the code yet. If the extraction of these IDs require an additional network request, it is better imo to just hardcode them since the brightcove IDs dont change

**Comment by coletdjnz** [comment]
> > I notice a few other similar extractors hard code the Brightcove player and account ids. Might be a bit overkill but I've added them as a fallback here and opted to try extract them first.
> 
> I havent looked at the code yet. If the extraction of these IDs require an additional network request, it is better imo to just hardcode them since the brightcove IDs dont change

Ah ok. I can simplify this a lot then.

File: yt_dlp/extractor/nzherald.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+from .brightcove import BrightcoveNewIE
+from ..compat import compat_str
+from .common import InfoExtractor
+from ..utils import (
+    ExtractorError,
+    traverse_obj
+)
```
Comment (pukkandan) [inline yt_dlp/extractor/nzherald.py:10]:
```suggestion
from .brightcove import BrightcoveNewIE
from .common import InfoExtractor

from ..compat import compat_str
from ..utils import (
    ExtractorError,
    traverse_obj
)
```

**Review by pukkandan — APPROVED** [review]
> raises unexpected extractor error for articles without a video

example URL? can't we detect and throw a proper error?

**Pull request merged**